src/mod_pushoff_fcm.erl

```diff
@@ -0,0 +1,336 @@
+                   retry_list = RetryList,
+                   pending_timer = PendingTimer,
+                   retry_timestamp = StoredTimestamp} = State) ->
+    case erlang:read_timer(PendingTimer) of
+        false -> self() ! send;
+        _ -> meh
+    end,
+    {noreply,
+     State#state{send_queue = lists:foldl(fun(E, Q) -> queue:snoc(Q, E) end, SendQ, RetryList),
+                 retry_list = []}};
+
+handle_info({retry, _T1}, #state{retry_timestamp = _T2} = State) ->
+    {noreply, State};
+
+handle_info({pending_timeout, Timestamp},
+            #state{pending_timestamp = StoredTimestamp} = State) ->
+    NewState =
+    case Timestamp of
+        StoredTimestamp ->
+            self() ! send,
+            State#state{pending_list = []};
+
+        _ -> State
+    end,
+    {noreply, NewState};
+
+handle_info(send, #state{pending_list = PendingList,
+                         send_queue = SendQ,
+                         retry_list = RetryList,
+                         retry_timer = RetryTimer,
+                         message_id = MessageId,
+                         api_key = ApiKey} = State) ->
+            {NewPendingList,
+             NewSendQ,
+             NewMessageId} = enqueue_some(PendingList, SendQ, MessageId),
+
+            Socket = {},
+            NewState = 
+            case NewPendingList of
+                [] ->
+                    State#state{pending_list = NewPendingList,
+                                send_queue = NewSendQ,
+                                message_id = NewMessageId};
+
+                _ ->
+
+                    HTTPOptions = [],
+                    Options = [],
+                    
+                    [Head | _] = NewPendingList,
+                    Body = case Head of 
+                        {_MessageId, {_, _Payload, _Token, _}} ->
+                            [_, {body, _Body}, {from, _From}] = _Payload,
+                            PushMessage = {struct, [
+                                                    {to,           _Token},
+                                                    {data,         {struct, [
+                                                                             {body, _Body},
+                                                                             {title, _From}
+                                                                            ]}},
+                                                    {notification, {struct, [
+                                                                             {body, _Body},
+                                                                             {title, _From}
+                                                                            ]}}
+                                                   ]},
+                            EncodedMessage = iolist_to_binary(mochijson2:encode(PushMessage)),
+                            ?DEBUG("AFTER mochijson2:encode() MESSAGE IS <~p>", [EncodedMessage]),
```

Comment:
please make sure you don't leak any message content into logs, even if these are for debugging